wire-ios-data-model/Source/Model/Message/ZMMessage+Reaction.swift

```diff
@@ -19,16 +19,32 @@
 import Foundation
 
 extension ZMMessage {
-    static func add(reaction: WireProtos.Reaction, senderID: UUID, conversation: ZMConversation, inContext moc: NSManagedObjectContext) {
+
+    static func add(
+        reaction: WireProtos.Reaction,
+        senderID: UUID,
+        conversation: ZMConversation,
+        inContext context: NSManagedObjectContext
+    ) {
         guard
-            let user = ZMUser.fetch(with: senderID, in: moc),
+            let user = ZMUser.fetch(with: senderID, in: context),
             let nonce = UUID(uuidString: reaction.messageID),
-            let localMessage = ZMMessage.fetch(withNonce: nonce, for: conversation, in: moc)
+            let localMessage = ZMMessage.fetch(withNonce: nonce, for: conversation, in: context)
         else {
             return
         }
 
-        localMessage.addReaction(reaction.emoji, forUser: user)
+        localMessage.setReactions(reaction.toSet(), forUser: user)
```

Comment:
`WireProtos.Reaction` comes from the protobuf definition, so we can't change it. Traditionally it contained only 0 or 1 reactions, but it has been repurposed to now contain 0 or more reactions. It should be called `Reactions`.

Otherwise, when there is reference to a "reaction", it generally means a single reaction, i.e a single emoji.

Reply:
Ok, since `WireProtos.Reaction` is not something we can easily change, could we somehow make it a little bit more explicit that we're mapping from the protobuf reaction to the message's reactions ?

Maybe something like this?

```swift
localMessage.setReactions(protobufReaction.toReactionsSet(), forUser: user)
```

Reply:
Done!